eksrollup/lib/k8s.py

```diff
@@ -152,6 +152,131 @@ def taint_node(node_name):
+
+        for pod in pods.items:
+
+            pod_name = pod.metadata.name
+            namespace = pod.metadata.namespace
+
+            # Using ownerReferences to verify the pod is owned by a ReplicaSet
+            for pod_owner_reference in pod.metadata.owner_references:
+                if pod_owner_reference.kind == "ReplicaSet":
+                    rs_name = pod_owner_reference.name
+
+                    try:
+                        # Get the ReplicaSet of the pod
+                        rs = apps_k8s_api.read_namespaced_replica_set(rs_name, namespace)
+
+                        # Only rollout restart single replica ReplicaSet
+                        if rs.status.replicas > 1:
+                            logger.debug('pod {} attached to ReplicaSet {} has more than 1 replica. Expecting node draining process to evict pod...'.format(pod_name, rs_name))
+                            continue
+
+                        # Catch if ReplicaSet is 0. Can happen if a deployment has been updated during the run
+                        if rs.status.replicas == 0:
+                            logger.debug('pod {} attached to ReplicaSet {} has 0 replicas...'.format(namespace, rs_name))
+                            continue
+
+                        # Using ownerReferences to get the deployment name
+                        for rs_owner_reference in rs.metadata.owner_references:
+                            if rs_owner_reference.kind == "Deployment":
+                                deployment_name = rs_owner_reference.name
+
+                                # Run the function in a thread to get concurrency
+                                logger.info('Restarting deployment {} in namespace {}...'.format(deployment_name, namespace))
+                                thread = threading.Thread(target=rolling_restart, args=(deployment_name, namespace))
+                                threads.append(thread)
+                                thread.start()
+
+
+                    except ApiException as e:
+                        logger.info("Exception when calling CoreV1Api->read_namespaced_replica_set: {}".format(e))
+
+    except ApiException as e:
+        logger.info("Exception when calling CoreV1Api->list_pod_for_all_namespaces: {}".format(e))
+
+    # Wait for each thread to join back into the main proccess
+    # Ensures threads complete before node draining proceeds
+    for thread in threads:
+        thread.join()
+
+
+def rolling_restart(deployment_name, namespace):
+    """
+    Using kubectl, perform a rolling restart of the deployment. The rolling
+    restart will ensure the new pod is marked as healthy before the old
+    starts to terminate.
+    """
+    # Using kubectl to perform a rollout restart.
+    kubectl_args = [
+        'kubectl', 'rollout', 'restart',
+        '--namespace', namespace,
+        'deployment', deployment_name
+    ]
+
+    if app_config['DRY_RUN'] is True:
```

Comment:
Is `DRY_RUN` always set even if it's false? This will raise if that key is missing from the dict. If it's always set and always boolean, then you don't need `is True`.

Reply:
Yeah, this is set in [config.py](https://github.com/mx51/eks-rolling-update/blob/90184c8a7638eec254d0bd12ad42b527aec18620/eksrollup/config.py#L28). Which will default to false.

Reply:
Yeah cool, in that case just `if app_config['DRY_RUN']:`

Reply:
Or if you ever need to do something when it's false `if not app_config['DRY_RUN']:`

Reply:
Awesome. made the update to it now. Thanks Mike!